File: ClothingWarehouseProjectFRONT/src/app/model/transaction.ts
import {Inventory} from "./inventory";
import {User} from "./user";

export class Transaction{

   id: number
   soldPrice: number
   transactionDate: string
   inventory: Inventory
   user: User


  constructor() {
    this.id = 0
    this.soldPrice = 0
    this.transactionDate = ""
    this.inventory = new Inventory()
    this.user = new User()
  }

}

File: ClothingWarehouseAPI/src/main/java/com/example/clothingwarehouseproject/domain/Transaction.java
package com.example.clothingwarehouseproject.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.sql.Date;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    private Date transactionDate;
    @OneToOne(fetch = FetchType.LAZY)
    private Inventory inventory;
    private double soldPrice;
    @OneToOne(fetch = FetchType.EAGER)
    private User user;

}
